index.js

```diff
@@ -342,7 +344,48 @@ app.post('/api/doorTest', async(req, res) => {
     const {locationid} = req.body;
     await redis.addDoorTestSensorData(req, res);
     await handleSensorRequest(locationid)
-});
+})
+
+app.post('/api/door', async(request, response) => {
+    const coreId = request.body.coreid
+    const location = await db.getLocationIDFromParticleCoreID(coreId)
+    const locationid = location.locationid
+  
+    const message = request.body.data
+    const signal = message.data
+    var doorSignal
+  
+    if (signal==IM21_DOOR_STATUS.OPEN){
+        doorSignal = SESSIONSTATE_DOOR.OPEN
+    }
+    else if (signal==IM21_DOOR_STATUS.CLOSED){
+        doorSignal = SESSIONSTATE_DOOR.CLOSED
+    }
+    else if (signal==IM21_DOOR_STATUS.LOW_BATT){
+        doorSignal = "LowBatt"
+    }
+    else if (signal==IM21_DOOR_STATUS.HEARTBEAT_OPEN || signal==IM21_DOOR_STATUS.HEARTBEAT_CLOSED){
+        doorSignal = "HeartBeat"
+    }
+  
+    redis.addIM21DoorSensorData(locationid, doorSignal)
+    handleSensorRequest(locationid)
+    response.status(200).json("OK")
+})
+
+// Handler for device vitals such as wifi strength
+app.post('/api/devicevitals', async(request, response) => {
+    const coreId = request.body.coreid
+    const location = await db.getLocationIDFromParticleCoreID(coreId)
+    const locationid = location.locationid
```

Comment:
Also need some error handling here for when `location` is `null`.

Reply:
good call